src/sail_json_backend/json.ml

```diff
@@ -398,14 +398,117 @@ let remove_identity_funcs op =
       else op
   ) op identity_funcs
 
+let extract_opt_operands_fn pattern op_list =
+    (* 
+       This extracts the optional operands function names from the captured regex:
+       eg: maybe_vmask(vm) -> maybe_vmask
+       This if further used as key to the mappings table to extract the optional
+       operands.
+    *)
+  let captured_operand =
+    List.find_map (fun s ->
+      if Str.string_match pattern s 0 then
+        let captured_group = Str.matched_group 0 s in
+        let optional_operand_parts = String.split_on_char '(' captured_group in
+        let captured_op = List.hd optional_operand_parts in
+        debug_print ("Captured operand: " ^ captured_op);
+        Some captured_op
+      else
+        None
+    ) op_list
```

Comment:
Given the size of this embedded function, I prefer moving it to be standalone.

Reply:
done.